backfill.sql

```diff
@@ -189,23 +193,35 @@ BEGIN
         -- find the dimension slices that overlap with the data in our staging table 
         -- the range_ends are non inclusive, the range_starts are inclusive
         AND max_time_internal >= ds.range_start AND min_time_internal < ds.range_end
+        ORDER BY ds.range_end
     LOOP
         -- decompress the chunks in the dimension slice, committing transactions after each decompress
         CALL decompress_dimension_slice(dimension_slice_row, chunks_decompressed);
 
```

Comment:
We could take out an explicit lock on the table here, to prevent the compression job from succeeding?

Reply:
Hmm...I think you just run into potential deadlock issues.